src/components/checkbox/checkbox.tsx

```diff
@@ -5,70 +5,37 @@ import {noop} from '../../utils';
 import styles from './checkbox.st.css';
 
 export interface CheckBoxProps extends FormInputProps<boolean>, properties.Props {
-    boxIcon?: React.ComponentType<CheckBoxIconProps>;
-    tickIcon?: React.ComponentType<CheckBoxIconProps>;
-    indeterminateIcon?: React.ComponentType<CheckBoxIconProps>;
+    tickIcon?: React.ReactNode;
+    indeterminateIcon?: React.ReactNode;
     children?: React.ReactNode;
     disabled?: boolean;
     readonly?: boolean;
+    error?: boolean;
```

Comment:
I suggest we add this only after there is a design for it...

Reply:
@JoMarton we have in wix theme